src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1032,34 +1083,51 @@ private void RemoveWorseMembers<TMember>(ArrayBuilder<MemberResolutionResult<TMe
             // then we need to do no more work; we know it cannot win. But it is also possible that
             // it is not worse than anything but not better than everything. 
 
+            if (results.Count <= 1)
```

Comment:
According to the comment above `RemainingCandidatesCount` these all need another data structure so if you don't mind I'd rather clean this up as part of that effort.